components/scream/src/physics/p3/p3_functions_check_values_impl.hpp

```diff
@@ -25,17 +25,17 @@ namespace p3 {
 template <typename S, typename D>
 KOKKOS_FUNCTION
 void Functions<S,D>
-::check_values(const uview_1d<Spack>& qv, const uview_1d<Spack>& temp, const Int& kts, const Int& kte,
+::check_values(const uview_1d<const Spack>& qv, const uview_1d<const Spack>& temp, const Int& ktop, const Int& kbot,
```

Comment:
Just a question here: what does a const template parameter buy us? There's language in the C++ standard that suggests that const qualifiers on these types of template parameters are ignored.

Reply:
Consider Kokkos array `const View<double*> A`. A is still writeable; the const-ness applies to the bulk data pointer to which A points. In contrast, `View<const double*> A` is not writeable. Templating doesn't change that; `qv` in the modification above is not writeable if `Spack` is `Pack<Real>`, whereas in the original code it was.

Reply:
I understand the intention of the const. What I'm saying is that I'm not convinced that a standards-compliant compiler respects this distinction. :-) Even so, I suppose it can be a nice reminder of the intention for the type.

Reply:
You could just try it in a small program. My guess is the const is active.

Reply:
```C++
#include <Kokkos_Core.hpp>
namespace ko = Kokkos;

template <typename T>
void foo (const Kokkos::View<T*>& a, const Kokkos::View<const T*>& b) {
  a[0] = 1;
  //b[0] = 2; // nope
}

int main () {
  Kokkos::View<double*> a("a", 10);
  Kokkos::View<const double*> b = a;
  foo(a, b);
}
```